### DISCUSSION TITLE
How to fire a request from a pre- or post-request script?

### DISCUSSION THREAD

**Discussion by igel1** [usebruno/bruno#1388] [Q&A]
Is there an easy way to run a request of your collection via pre- or post-request scripting?
(something like "bru.run("name of request")

I know, that I can run an arbitrary request via bru CLI, but I do not know, how to
run a request from a pre- or post-request script out of an existing request.
(I have to confess: I do have only very little JavaSkript knowledge)

Kind regards

Igel1

**Comment by rishacha** [comment]
I also have a similar scenario where I have a pre-request script to get the JWT Token and store it in an environment variable.

Example - https://rakhayyat.medium.com/using-postman-pre-request-script-for-jwt-authentication-bcad415f8171#:~:text=Pre%2Drequest%20scripts%20are%20snippets,value%20returned%20from%20a%20function.

```js
const loginUrl = pm.environment.get("baseUrl") + "login";
pm.sendRequest({
    url: loginUrl,
    method: 'POST',
    header: {
        'content-type': 'application/json'
    },
    body: {
        mode: 'raw',
        raw: JSON.stringify({"username": "admin",
	                         "password": "password" })
        
    }
}, function (err, res) {
    pm.environment.set("auth",  res.json().token);
});
```

**Reply by rishacha** [reply]
Perhaps there's an answer in this thread - #535

**Comment by igel1** [comment]
Thanks for your efforts, Rishabh.

I know how these things are done in Postman and I know, how to implement script based access-token handling in Bruno - I have implemented these things by myself. So I also know about the axios ceremony.

But this time I really want to run an arbitrary request from my existing collection.
And I want to start this request (including it's pre- and post-request scripts) 
from a pre- or post-request script of an existing request, which also belongs to the same collection.

It is exactly the feature, that scflode asked for in your quoted thread (https://github.com/usebruno/bruno/discussions/535):

"The point of the idea here is if it's possible to have something like bru.run("Request name") basically wrapping the Axios ceremony."

Anoop responded: "Yes, We will support this soon."
Do you know, where I can track the progress of this feature-request?
